src/codeCache.h

```diff
@@ -168,6 +168,10 @@ class CodeCache {
         return _max_address;
     }
 
+    const char* imageBase() const {
+        return _image_base != NULL ? _image_base : (char*) minAddress();
```

Comment:
Does this cover all cases in `parseLibraries`?

Reply:
The only case when this may not behave exactly like the previous implementation is for `[vdso]`:
```
(gdb) print lib.file
$1 = 0x55555555ea60 "[vdso]"
(gdb) print lib.image_base
$2 = 0x7ffff7fc3000 "\177ELF\002\001\001"
(gdb) print lib.map_start
$3 = 0x7ffff7fc3000 "\177ELF\002\001\001"
```
Luckily in this case `lib.image_base == lib.map_start`, is this something we can rely on @krk ?

Reply:
I don't think so.
But why is this change needed in the first place? A function called `imageBase()` that may return something other than image base looks suspicious to me.

Reply:
So maybe we can't just use the `image_base` we store in `CodeCache` as a full replacement of the second argument of the second argument of `parseFile` and `parseProgramHeaders`. I reverted the commit and removed the TODO which triggered this change: bc306667edf00872fe74766694c27da3e1e05068